pkg/remote/proxy.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"context"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/api/httpbody"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// Proxy implements a local proxy for a remote Registry server.
+// This allows tests written to the generated "RegistryServer" and "AdminServer" interfaces to be run against remote servers.
+type Proxy struct {
+	adminClient    connection.AdminClient
+	registryClient connection.Client
+
+	rpc.UnimplementedRegistryServer
+	rpc.UnimplementedAdminServer
+}
+
+func (p *Proxy) Open(ctx context.Context) error {
+	var err error
+	p.adminClient, err = connection.NewAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Delete everything in the remote server to be tested.
+	it := p.adminClient.ListProjects(ctx, &rpc.ListProjectsRequest{})
```

Comment:
Deleting everything in the remote server would be an astonishing side-effect of an Open() function (at least to me). We should make the delete an explicit call or perhaps change the name of the function to indicate this behavior as well as add a comment to that effect. (All public functions should have comments anyway.)

Reply:
For sure, yes, that's putting it politely. In addition to warning people about this, I wonder if we could limit its blast radius by adopting some testing conventions -- if all projects used in testing matched a certain pattern (e.g. `".*test.*"`), then we could only clear those projects and not others, though this might still break a few tests that enumerate all the projects in a running server.